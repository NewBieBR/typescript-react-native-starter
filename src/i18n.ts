import i18n from 'i18next';
import {initReactI18next} from 'react-i18next';

export const defaultNS = 'translation';

export const resources = {
  en: {
    translation: {
      home: 'Home',
      settings: 'Settings',
    },
  },
  fr: {
    translation: {
      home: 'Accueil',
      settings: 'Param√®tres',
    },
  },
} as const;

i18n.use(initReactI18next).init({
  ns: ['translation'],
  defaultNS,
  resources,
  lng: 'en',
  keySeparator: '.',
  interpolation: {
    escapeValue: false, // react already safes from xss
  },
  compatibilityJSON: 'v3', // hermes compatible
});

export default i18n;
