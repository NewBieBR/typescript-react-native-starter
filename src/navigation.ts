import {AppNavigations, AppStackRoutes} from 'src/navigators/AppStack';
import {MainBottomTabNavigations, MainBottomTabRoutes} from 'src/navigators/MainBottomTab';

export type Navigations = AppNavigations & MainBottomTabNavigations;

export type Routes = AppStackRoutes & MainBottomTabRoutes;
