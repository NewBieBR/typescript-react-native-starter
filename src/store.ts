import AsyncStorage from '@react-native-async-storage/async-storage';
import {configureStore} from '@reduxjs/toolkit';
import {applyMiddleware, combineReducers, compose, Dispatch, MiddlewareAPI} from 'redux';
import {persistReducer, persistStore} from 'redux-persist';
import createSagaMiddleware from 'redux-saga';
import {appReducer} from 'src/modules/app/reducer';
import {RootAction} from './types/actions';

/*
 *--------------------------------------------------*
 * Persist config documentation
 * https://github.com/rt2zz/redux-persist/blob/master/src/types.js#L13-L27
 *--------------------------------------------------*
 */

const appPersistConfig = {
  storage: AsyncStorage,
  key: 'app',
};

export const reducers = {
  app: persistReducer(appPersistConfig, appReducer),
};

export const rootReducer = combineReducers(reducers);

// eslint-disable-next-line @typescript-eslint/no-unused-vars
const appMiddleware = (_store: MiddlewareAPI) => (next: Dispatch) => (action: RootAction) => {
  //   var state = store.getState()
  //   switch (action.type) {
  //     case actions.ADD_TASK:
  //       *do something*
  //       break;
  //   }
  next(action);
};

const sagaMiddleware = createSagaMiddleware();

const middlewares = [sagaMiddleware, appMiddleware];
const enhancers = [applyMiddleware(...middlewares)];

export const store = configureStore({
  reducer: rootReducer,
  enhancers: compose(...enhancers),
});

export type RootState = ReturnType<typeof store.getState>;
export type AppDispatch = typeof store.dispatch;

// sagaMiddleware.run(sagas);

export const persistor = persistStore(store);

/*
 *--------------------------------------------------*
 * Reset persist store: persistor.purge()
 *--------------------------------------------------*
 */
