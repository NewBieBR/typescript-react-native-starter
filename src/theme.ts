import {StatusBarStyle} from 'react-native';
import {Colors} from 'src/constants/colors';

const CommonTheme = {
  gridUnit: 4,
  fontFamily: 'Helvetica',
};

export const LightTheme = {
  barStyle: 'dark-content' as StatusBarStyle,
  colors: {
    background: Colors.offwhite,
    text: Colors.black,
    ...Colors,
  },
  ...CommonTheme,
};

export const DefaultTheme = LightTheme;

export const DarkTheme = {
  barStyle: 'light-content' as StatusBarStyle,
  colors: {
    background: Colors.black,
    text: Colors.offwhite,
    ...Colors,
  },
  ...CommonTheme,
};

export type ThemeType = 'light' | 'dark' | 'system';

export const Themes: Record<ThemeType, typeof DefaultTheme> = {
  light: LightTheme,
  dark: DarkTheme,
  system: DefaultTheme,
};
