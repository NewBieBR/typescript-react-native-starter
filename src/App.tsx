import {NavigationContainer} from '@react-navigation/native';
import React, {memo, ReactNode} from 'react';
import {StatusBar} from 'react-native';
import {Provider} from 'react-redux';
import {AppStack} from 'src/navigators/AppStack';
import {store} from 'src/store';
import {ThemeProvider} from 'styled-components';
import {useSelector} from './hooks/useSelector';
import {themeTypeSelector} from './modules/app/selectors';
import {useColorScheme} from 'react-native';
import {Themes} from './theme';

const AppThemeProvider = ({children}: {children: ReactNode | ReactNode[]}) => {
  const userSelectedThemeType = useSelector(themeTypeSelector);
  const systemThemeType = useColorScheme();
  const themeType =
    userSelectedThemeType === 'system' && systemThemeType ? systemThemeType : userSelectedThemeType;
  const theme = Themes[themeType];

  return (
    <ThemeProvider theme={theme}>
      <StatusBar barStyle={theme.barStyle} />
      {children}
    </ThemeProvider>
  );
};

const AppComponent = (): JSX.Element => {
  return (
    <Provider store={store}>
      <AppThemeProvider>
        <NavigationContainer>
          <AppStack />
        </NavigationContainer>
      </AppThemeProvider>
    </Provider>
  );
};

export const App = memo(AppComponent);
