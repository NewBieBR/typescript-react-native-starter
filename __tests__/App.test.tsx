import {render} from '@testing-library/react-native';
import React from 'react';
import {App} from 'src/App';

describe('App', () => {
  it('should renders correctly', () => {
    const page = render(<App />);
    expect(page.toJSON()).toMatchSnapshot();
  });
});
