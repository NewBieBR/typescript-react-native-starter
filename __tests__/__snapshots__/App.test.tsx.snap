// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App should renders correctly 1`] = `
<RNCSafeAreaProvider
  onInsetsChange={[Function]}
  style={
    Array [
      Object {
        "flex": 1,
      },
      undefined,
    ]
  }
>
  <RNSScreenStack
    style={
      Object {
        "flex": 1,
      }
    }
  >
    <RNSScreen
      collapsable={false}
      forwardedRef={[Function]}
      onAppear={[Function]}
      onDisappear={[Function]}
      onDismissed={[Function]}
      onTransitionProgress={[Function]}
      onWillDisappear={[Function]}
      replaceAnimation="push"
      stackPresentation="push"
      style={
        Object {
          "bottom": 0,
          "left": 0,
          "position": "absolute",
          "right": 0,
          "top": 0,
        }
      }
    >
      <RNSScreenStackHeaderConfig
        backButtonInCustomView={false}
        backgroundColor="rgb(255, 255, 255)"
        color="rgb(0, 122, 255)"
        direction="ltr"
        disableBackButtonMenu={false}
        hidden={true}
        hideBackButton={false}
        largeTitleHideShadow={false}
        title="mainBottomTab"
        titleColor="rgb(28, 28, 30)"
        topInsetEnabled={false}
        translucent={false}
      />
      <View
        style={
          Array [
            Object {
              "flex": 1,
            },
            Object {
              "backgroundColor": "rgb(242, 242, 242)",
            },
            undefined,
          ]
        }
      >
        <View
          style={
            Array [
              Object {
                "flex": 1,
              },
              undefined,
            ]
          }
        >
          <RNSScreenNavigationContainer
            style={
              Object {
                "flex": 1,
                "overflow": "hidden",
              }
            }
          >
            <RNSScreen
              activityState={2}
              collapsable={false}
              forwardedRef={[Function]}
              style={
                Object {
                  "bottom": 0,
                  "left": 0,
                  "position": "absolute",
                  "right": 0,
                  "top": 0,
                  "zIndex": 0,
                }
              }
            >
              <View
                accessibilityElementsHidden={false}
                importantForAccessibility="auto"
                style={
                  Array [
                    Object {
                      "backgroundColor": "rgb(242, 242, 242)",
                      "flex": 1,
                    },
                    Array [
                      Object {
                        "flex": 1,
                        "flexDirection": "column-reverse",
                      },
                      undefined,
                    ],
                  ]
                }
              >
                <View
                  style={
                    Object {
                      "flex": 1,
                    }
                  }
                >
                  <View
                    className="sc-bczRLJ crvKQZ"
                  >
                    <Text
                      className="sc-gsnTZi iGbgVU"
                    >
                      home
                    </Text>
                  </View>
                </View>
              </View>
            </RNSScreen>
          </RNSScreenNavigationContainer>
          <View
            collapsable={false}
            onLayout={[Function]}
            pointerEvents="auto"
            style={
              Object {
                "backgroundColor": "#f4f5f6",
                "borderTopColor": "#f4f5f6",
                "borderTopWidth": 0.5,
                "bottom": 0,
                "elevation": 8,
                "height": 49,
                "left": 0,
                "paddingBottom": 0,
                "paddingHorizontal": 0,
                "position": null,
                "right": 0,
                "transform": Array [
                  Object {
                    "translateY": 0,
                  },
                ],
              }
            }
          >
            <View
              pointerEvents="none"
              style={
                Object {
                  "bottom": 0,
                  "left": 0,
                  "position": "absolute",
                  "right": 0,
                  "top": 0,
                }
              }
            />
            <View
              accessibilityRole="tablist"
              style={
                Object {
                  "flex": 1,
                  "flexDirection": "row",
                }
              }
            >
              <View
                accessibilityLabel="home, tab, 1 of 2"
                accessibilityRole="button"
                accessibilityState={
                  Object {
                    "selected": true,
                  }
                }
                accessibilityStates={
                  Array [
                    "selected",
                  ]
                }
                accessible={true}
                collapsable={false}
                focusable={true}
                onBlur={[Function]}
                onClick={[Function]}
                onFocus={[Function]}
                onResponderGrant={[Function]}
                onResponderMove={[Function]}
                onResponderRelease={[Function]}
                onResponderTerminate={[Function]}
                onResponderTerminationRequest={[Function]}
                onStartShouldSetResponder={[Function]}
                style={
                  Array [
                    Object {
                      "alignItems": "center",
                      "flex": 1,
                    },
                    Object {
                      "backgroundColor": "transparent",
                    },
                    Object {
                      "flexDirection": "column",
                      "justifyContent": "flex-end",
                    },
                    undefined,
                  ]
                }
              >
                <View
                  style={
                    Array [
                      Object {
                        "flex": 1,
                      },
                      undefined,
                    ]
                  }
                >
                  <View
                    style={
                      Array [
                        Object {
                          "alignItems": "center",
                          "alignSelf": "center",
                          "height": "100%",
                          "justifyContent": "center",
                          "minWidth": 25,
                          "position": "absolute",
                          "width": "100%",
                        },
                        Object {
                          "opacity": 1,
                        },
                      ]
                    }
                  >
                    <Text
                      allowFontScaling={false}
                      selectable={false}
                      style={
                        Array [
                          Object {
                            "color": "rgb(0, 122, 255)",
                            "fontSize": 25,
                          },
                          undefined,
                          Object {
                            "fontFamily": "anticon",
                            "fontStyle": "normal",
                            "fontWeight": "normal",
                          },
                          Object {},
                        ]
                      }
                    >
                      
                    </Text>
                  </View>
                  <View
                    style={
                      Array [
                        Object {
                          "alignItems": "center",
                          "alignSelf": "center",
                          "height": "100%",
                          "justifyContent": "center",
                          "minWidth": 25,
                          "position": "absolute",
                          "width": "100%",
                        },
                        Object {
                          "opacity": 0,
                        },
                      ]
                    }
                  >
                    <Text
                      allowFontScaling={false}
                      selectable={false}
                      style={
                        Array [
                          Object {
                            "color": "#8E8E8F",
                            "fontSize": 25,
                          },
                          undefined,
                          Object {
                            "fontFamily": "anticon",
                            "fontStyle": "normal",
                            "fontWeight": "normal",
                          },
                          Object {},
                        ]
                      }
                    >
                      
                    </Text>
                  </View>
                </View>
                <Text
                  numberOfLines={1}
                  style={
                    Array [
                      Object {
                        "backgroundColor": "transparent",
                        "textAlign": "center",
                      },
                      Object {
                        "color": "rgb(0, 122, 255)",
                      },
                      Object {
                        "fontSize": 10,
                      },
                      undefined,
                    ]
                  }
                >
                  home
                </Text>
              </View>
              <View
                accessibilityLabel="settings, tab, 2 of 2"
                accessibilityRole="button"
                accessibilityState={
                  Object {
                    "selected": false,
                  }
                }
                accessibilityStates={Array []}
                accessible={true}
                collapsable={false}
                focusable={true}
                onBlur={[Function]}
                onClick={[Function]}
                onFocus={[Function]}
                onResponderGrant={[Function]}
                onResponderMove={[Function]}
                onResponderRelease={[Function]}
                onResponderTerminate={[Function]}
                onResponderTerminationRequest={[Function]}
                onStartShouldSetResponder={[Function]}
                style={
                  Array [
                    Object {
                      "alignItems": "center",
                      "flex": 1,
                    },
                    Object {
                      "backgroundColor": "transparent",
                    },
                    Object {
                      "flexDirection": "column",
                      "justifyContent": "flex-end",
                    },
                    undefined,
                  ]
                }
              >
                <View
                  style={
                    Array [
                      Object {
                        "flex": 1,
                      },
                      undefined,
                    ]
                  }
                >
                  <View
                    style={
                      Array [
                        Object {
                          "alignItems": "center",
                          "alignSelf": "center",
                          "height": "100%",
                          "justifyContent": "center",
                          "minWidth": 25,
                          "position": "absolute",
                          "width": "100%",
                        },
                        Object {
                          "opacity": 0,
                        },
                      ]
                    }
                  >
                    <Text
                      allowFontScaling={false}
                      selectable={false}
                      style={
                        Array [
                          Object {
                            "color": "rgb(0, 122, 255)",
                            "fontSize": 25,
                          },
                          undefined,
                          Object {
                            "fontFamily": "anticon",
                            "fontStyle": "normal",
                            "fontWeight": "normal",
                          },
                          Object {},
                        ]
                      }
                    >
                      
                    </Text>
                  </View>
                  <View
                    style={
                      Array [
                        Object {
                          "alignItems": "center",
                          "alignSelf": "center",
                          "height": "100%",
                          "justifyContent": "center",
                          "minWidth": 25,
                          "position": "absolute",
                          "width": "100%",
                        },
                        Object {
                          "opacity": 1,
                        },
                      ]
                    }
                  >
                    <Text
                      allowFontScaling={false}
                      selectable={false}
                      style={
                        Array [
                          Object {
                            "color": "#8E8E8F",
                            "fontSize": 25,
                          },
                          undefined,
                          Object {
                            "fontFamily": "anticon",
                            "fontStyle": "normal",
                            "fontWeight": "normal",
                          },
                          Object {},
                        ]
                      }
                    >
                      
                    </Text>
                  </View>
                </View>
                <Text
                  numberOfLines={1}
                  style={
                    Array [
                      Object {
                        "backgroundColor": "transparent",
                        "textAlign": "center",
                      },
                      Object {
                        "color": "#8E8E8F",
                      },
                      Object {
                        "fontSize": 10,
                      },
                      undefined,
                    ]
                  }
                >
                  settings
                </Text>
              </View>
            </View>
          </View>
        </View>
      </View>
    </RNSScreen>
  </RNSScreenStack>
</RNCSafeAreaProvider>
`;
